File: framework/scala/src/main/scala/soda/scala/web/bootstrap/BootstrapContext.scala
package soda.scala.web.bootstrap

import soda.scala.web.WebUtils

class BootstrapContext(private val key: String, artifact: Array[Byte]) {

  private var md5: String = ""

  private var classLoader: ClassLoader = null

  private def init(): Unit = {
    md5 = WebUtils.md5Hex(artifact)
    val parentLoader = Thread.currentThread().getContextClassLoader
    classLoader = new JarFileBytesClassLoader(artifact, parentLoader)
  }

  def getKey: String = key

  def getMd5: String = md5

  def getClassLoader: ClassLoader = classLoader

  init()

}

File: framework/kotlin/src/main/kotlin/soda/kotlin/web/bootstrap/BootstrapContext.kt
package soda.kotlin.web.bootstrap

import soda.kotlin.web.WebUtils

class BootstrapContext(private val key: String, artifact: ByteArray) {

    private var md5: String

    private var classLoader: ClassLoader

    init {
        md5 = WebUtils.md5Hex(artifact)
        val parentLoader = Thread.currentThread().contextClassLoader
        classLoader = JarFileBytesClassLoader(artifact, parentLoader)
    }

    fun getKey() = key

    fun getMd5() = md5

    fun getClassLoader() = classLoader

}